pkg/apiserver/webhooks/datavolume-mutate.go

```diff
@@ -127,6 +127,11 @@ func (wh *dataVolumeMutatingWebhook) Admit(ar admissionv1.AdmissionReview) *admi
 		sourceNamespace = targetNamespace
 	}
 
+	_, err = wh.k8sClient.CoreV1().Namespaces().Get(context.TODO(), sourceNamespace, metav1.GetOptions{})
```

Comment:
I also thought that, but I tested several cases where the source namespace didn't exist and `clone.CanUserClonePVC` always succeeded. Mind that, before my PR, all related unit tests also succeeded without having the namespaces, so that's why I had to mock these after adding the code above:

https://github.com/kubevirt/containerized-data-importer/blob/43cc9d3e695a410addb7ce1449123fe8c29f2507/pkg/apiserver/webhooks/datavolume-mutate_test.go#L309-L310

Reply:
So remember in the unit tests we don't have a full fledged API server, so it is unlikely to fail on something like that. Does it not reject in functional tests is the question here. If it does, we need to ensure that we get the same behavior in the unit tests.